include/mplib/core/articulated_model.h

```diff
@@ -186,18 +187,21 @@ class ArticulatedModelTpl {
   void setQpos(const VectorX<S> &qpos, bool full = false);
 
   /**
-   * Get the base pose of the robot.
+   * Get the base (root) pose of the robot.
    *
-   * @return: base pose of the robot in [x, y, z, qw, qx, qy, qz] format
+   * @return: base pose of the robot
    */
-  Vector7<S> getBasePose() const { return toPoseVec<S>(base_pose_); }
+  Pose<S> getBasePose() const { return Pose<S>(base_pose_); }
 
   /**
-   * Set the base pose of the robot.
+   * Set the base pose of the robot and update all collision links in the ``FCLModel``.
    *
-   * @param pose: base pose of the robot in [x, y, z, qw, qx, qy, qz] format
+   * @param pose: base pose of the robot
    */
-  void setBasePose(const Vector7<S> &pose);
+  void setBasePose(const Pose<S> &pose) {
+    base_pose_ = pose.toIsometry();
```

Comment:
Yes, this is intended as I stated in `mplib::pose` docstring:

>  Pose stored as a unit quaternion and a position vector
    This struct is intended to be used only for interfacing with Python.
    Internally, ``Pose`` is converted to and stored as ``Eigen::Isometry3``
    which is used by all computations.

This is simply because doing computations in Pose will eventually be needed to convert to Isometry3. So it's better to just take the cost during construction / set_pose, etc.